adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala

```diff
@@ -300,45 +270,368 @@ private[adam] class VariantContextConverter(
+        })
+      }).getOrElse(vcb.unfiltered)
+  }
+
+  private val variantExtractFns: Iterable[(Variant, VariantContextBuilder) => VariantContextBuilder] = Iterable(
+    extractNames(_, _),
+    extractFilters(_, _)
+  )
+
+  // htsjdk --> variant annotation format functions
+
+  private[converters] def formatAncestralAllele(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttributeAsString("AA", null))
+      .fold(vab)(vab.setAncestralAllele(_))
+  }
+
+  private[converters] def formatDbSnp(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("DB").asInstanceOf[java.lang.Boolean])
+      .fold(vab)(vab.setDbSnp(_))
+  }
+
+  private[converters] def formatHapMap2(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("H2").asInstanceOf[java.lang.Boolean])
+      .fold(vab)(vab.setHapMap2(_))
+  }
+
+  private[converters] def formatHapMap3(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("H3").asInstanceOf[java.lang.Boolean])
+      .fold(vab)(vab.setHapMap3(_))
+  }
+
+  private[converters] def formatValidated(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("VALIDATED").asInstanceOf[java.lang.Boolean])
+      .fold(vab)(vab.setValidated(_))
+  }
+
+  private[converters] def formatThousandGenomes(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("1000G").asInstanceOf[java.lang.Boolean])
+      .fold(vab)(vab.setThousandGenomes(_))
+  }
+
+  private[converters] def formatSomatic(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("SOMATIC").asInstanceOf[java.lang.Boolean])
+      .fold(vab)(vab.setSomatic(_))
+  }
+
+  private[converters] def formatAlleleCount(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    val ac = vc.getAttributeAsList("AC")
+    if (ac.size > index) {
+      vab.setAlleleCount(toInt(ac.get(index)))
+    }
+    vab
+  }
+
+  private[converters] def formatAlleleFrequency(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    val af = vc.getAttributeAsList("AF")
+    if (af.size > index) {
```

Comment:
Not sure I'm following you, if the `af.size` is zero then we don't want to set the value